src/test/kotlin/se/uulm/snowballr/backend/repository/association/ProjectMemberTableRepoTest.kt

```diff
@@ -260,4 +263,35 @@ class ProjectMemberTableRepoTest : RepositoryTest(arrayOf(ProjectTable, ProjectM
             assertThat(promotedMember.role).isEqualTo(MemberRole.MEMBER_ROLE_ADMIN)
         }
     }
+
+    @Nested
+    inner class GetProjectMembersWithUsers {
+        @Test
+        fun `When a project member and the corresponding user exists, then the project member with user is correctly returned`() =
+            runTest {
+                val (project, _) = setupProject()
+                val normalMember = repo.getProjectMemberByComposedId(project.id, testUserId)
+                val user = userRepo.getUserById(testUserId)
+                val projectMembersWithUsers = repo.getProjectMembersWithUsers(project.id)
+
+                assertThat(projectMembersWithUsers).hasSize(1)
+                assertThat(projectMembersWithUsers).anyMatch { it.projectMember == normalMember }
+                assertThat(projectMembersWithUsers).anyMatch { it.user == user }
+            }
+
+        @Test
+        fun `When a not all users are project members, then only the correct project members with users are returned`() =
+            runTest {
+                val (project, _) = setupProject()
+                val user = userRepo.getUserById(testUserId)
+                val nonProjectMemberUser = DataBuilder.createExampleUser()
```

Comment:
This only creates a data object. Use the repo helper to actually persist a user.
```suggestion
                val nonProjectMemberUserId = RepositoryHelper.createExampleUser("test-user@example.com")
```